### ISSUE TITLE
Adding services to an existing machine causes inconsistent results

### ISSUE THREAD

**Issue by jonahgeorge** [fly-apps/terraform-provider-fly#77]
Hey @DAlperin! Thanks a ton for creating and maintaining this provider. I'm very excited to increase usage of it. I've been having some non-starters, but after upgrading to 0.15 I got my first successful app launches. 

Now, when trying to add services to these existing apps I'm running into some consistency issues. I've tried to reduce the problem as much as possible and include all available debug info. Let me know if there's anything else that would be helpful! 

It's worth noting that entirely recreating the app does not cause this same issue.

```sh
terraform apply
fly_app.nginx: Refreshing state... [id=jg-nginx]
fly_machine.nginx: Refreshing state... [id=e784343a275778]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # fly_machine.nginx will be updated in-place
  ~ resource "fly_machine" "nginx" {
      ~ env      = {} -> (known after apply)
      ~ id       = "e784343a275778" -> (known after apply)
        name     = "jg-nginx"
      + services = [
          + {
              + internal_port = 80
              + ports         = [
                  + {
                      + handlers = [
                          + "http",
                        ]
                      + port     = 80
                    },
                ]
              + protocol      = "tcp"
            },
        ]
        # (6 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fly_machine.nginx: Modifying... [id=e784343a275778]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to fly_machine.nginx, provider "provider[\"registry.terraform.io/fly-apps/fly\"]" produced an unexpected new value: .services: was
│ cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"internal_port":cty.NumberIntVal(80),
│ "ports":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"handlers":cty.ListVal([]cty.Value{cty.StringVal("http")}), "port":cty.NumberIntVal(80)})}), "protocol":cty.StringVal("tcp")})}), but
│ now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

---

[`DEBUG=1 TF_LOG=debug terraform apply`](https://gist.github.com/jonahgeorge/41fd76a1189dd8621720a6f7c1943f00)

<details>
<summary>Terraform Manifest</summary>

```terraform
terraform {
  required_providers {
    fly = {
      source  = "fly-apps/fly"
      version = "0.0.15"
    }
  }
}

provider "fly" {}

resource "fly_app" "nginx" {
  name = "jg-nginx"
  org  = "personal"
}

resource "fly_machine" "nginx" {
  name = "jg-nginx"

  app   = fly_app.nginx.name
  image = "nginx"

  region   = "sea"
  cpus     = 1
  memorymb = 1024
  cputype  = "shared"

  # Newly Added 
  services = [
    {
      internal_port = 80
      ports = [
        {
          handlers = [
            "http",
          ]
          port = 80
        },
      ]
      protocol = "tcp"
    }
  ]
}
```

</details>

**Comment by DAlperin** [comment]
Oops good catch. Solved in v0.0.16

**Comment by jonahgeorge** [comment]
Thank you!
